src/components/toaster/toaster-context.tsx

```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext, useState } from 'react';
+
+const TOAST_REMOVE_DELAY = 3000;
+type ToastType = 'success' | 'error' | 'warning' | 'info';
+
+interface ToastData {
+    type: ToastType;
+    message: string;
+    isOpen: boolean;
+    timeoutId: number;
+}
+
+export function ToasterContextProvider({ children }: React.PropsWithChildren) {
+    const [toastData, setToastData] = useState<ToastData | null>(null);
+
+    const toast = ({
+        type,
+        message,
+        duration,
+    }: {
+        type: ToastType;
+        message: string;
+        duration?: number;
+    }) => {
+        setToastData({
+            type,
+            message,
+            timeoutId: window.setTimeout(closeToast, duration ?? TOAST_REMOVE_DELAY),
+            isOpen: true,
+        });
+    };
+
+    const closeToast = () => {
+        setToastData((toast) => {
+            if (toast && toast.timeoutId) window.clearTimeout(toast.timeoutId);
+            return null;
+        });
+    };
+
+    const onOpenChange = () => (isOpen: boolean) => {
```

Comment:
the type of this is `() => (isOpen: boolean)  => void` but the context type says `onOpenChange: (isOpen: boolean) => void;`